generators/entity-server/templates/src/main/java/package/repository/_EntityRepository.java

```diff
@@ -76,9 +78,10 @@
     List<<%= entityClass %>> findBy<%= relationships[idx].relationshipNameCapitalized %>IsCurrentUser();
     <%_ } } _%>
     <%_ if (fieldsContainOwnerManyToMany === true) { _%>
-    @Query("select distinct <%= entityTableName %> from <%= entityClass %> <%= entityTableName %><% for (idx in relationships) {
-    if (relationships[idx].relationshipType === 'many-to-many' && relationships[idx].ownerSide === true) { %> left join fetch <%=entityTableName%>.<%=relationships[idx].relationshipFieldNamePlural%><%} }%>")
-    List<<%=entityClass%>> findAllWithEagerRelationships();
+    @Query(value = "select distinct <%= entityTableName %> from <%= entityClass %> <%= entityTableName %><% for (idx in relationships) {
+    if (relationships[idx].relationshipType === 'many-to-many' && relationships[idx].ownerSide === true) { %> left join fetch <%=entityTableName%>.<%=relationships[idx].relationshipFieldNamePlural%>",
+        countQuery = "select count(distinct <%= entityTableName %>) from <%= entityClass %> <%= entityTableName %><%} }%>")
+    Page<<%=entityClass%>> findAllWithEagerRelationships(Pageable pageable);
```

Comment:
need to fix the call of this function as well in `_EntityServiceImpl.java` and `get_all_template.ejs`, and passing `null` (if I'm not wrong)

Reply:
yes, I marked this PR as WIP for the time being.